test/e2e/self_hosted.go

```diff
@@ -73,11 +86,27 @@ func SelfHostedSpec(ctx context.Context, inputGetter func() SelfHostedSpecInput)
 		Expect(input.ClusterctlConfigPath).To(BeAnExistingFile(), "Invalid argument. input.ClusterctlConfigPath must be an existing file when calling %s spec", specName)
 		Expect(input.BootstrapClusterProxy).ToNot(BeNil(), "Invalid argument. input.BootstrapClusterProxy can't be nil when calling %s spec", specName)
 		Expect(os.MkdirAll(input.ArtifactFolder, 0750)).To(Succeed(), "Invalid argument. input.ArtifactFolder can't be created for %s spec", specName)
-		Expect(input.E2EConfig.Variables).To(HaveKey(KubernetesVersion))
+		Expect(input.E2EConfig.Variables).To(HaveKey(KubernetesVersionUpgradeFrom))
```

Comment:
This makes it a breaking change. But I assume that's okay for tests (?)

cc @fabriziopandini

Reply:
Should be not-breaking anymore.
The whole upgrade is now toggle-able via the `EnableUpgrade` bool, which defaults to false (and thus preserves the old behaviour)

Reply:
Sounds good, resolving for now (I'll do another review when ready)